includes/wikia/models/DesignSystemGlobalNavigationModelV2.class.php

```diff
@@ -92,11 +92,64 @@ private function getCreateWiki( string $trackingLabel ) {
 		];
 	}
 
-	private function getHref( $hrefKey, $protocolRelative = false ) {
+	protected function getDomainForCityId( $cityId ) {
+		global $wgServer;
+		if ( $cityId == null ) {
+			return $wgServer;
+		} else {
+			return  WikiFactory::cityIDtoUrl( $cityId );
+		}
+	}
+
+	protected function getDevDomain() {
+		global $wgDevDomain, $wgDevelEnvironment;
+		if ( !$wgDevelEnvironment || empty( $wgDevDomain ) ) {
+			return false;
+		}
+
+		return $wgDevDomain;
+	}
+
+	public function usePrimaryDomainInUrl( $url, $cityId = null ) {
+		global $wgWikiaBaseDomainRegex;
+
+		$cityUrl = $this->getDomainForCityId( $cityId );
+		// Should probably check for HTTPS in some cases (for now all processed links should be HTTPS or relative
+
+		$urlParts = [];
+		$cityUrlParts = [];
+		if (
+			preg_match( '/' . $wgWikiaBaseDomainRegex . '/', $url, $urlParts ) === 0 ||
+			preg_match( '/'. $wgWikiaBaseDomainRegex . '/', $cityUrl, $cityUrlParts ) === 0 ||
+			$urlParts[1] === $cityUrlParts[1]
+		) {
+			return $url;
+		}
+
+		$count = 0;
+		$devDomain = $this->getDevDomain();
+
+		if ( $devDomain ) {
+			$finalUrl = preg_replace( '/' . $wgWikiaBaseDomainRegex . '/', $devDomain, $url, 1, $count );
+		} else {
+			$finalUrl = preg_replace( '/' . $urlParts[1] . '/', $cityUrlParts[1], $url, 1, $count );
+		}
+
+		if ( $finalUrl == null || $count !== 1 ) {
```

Comment:
When would this happen? Would be nice to have a comment explaining it.

Reply:
It is highly unlikely to happen since the values used are captured from the regexp few lines before but function can fail and ignoring its results is not good idea.